### PR TITLE
add utf8_wchar

### PR DESCRIPTION
**Pull request by pavel-pimenov**
https://github.com/arvidn/libtorrent/issues/4361

### CODE REVIEW

File: src/posix_storage.cpp [hunk 1]
```diff
@@ -542,7 +553,11 @@ namespace aux {
 				// and make sure we create the file this time ("r+") opens for
 				// reading and writing, but doesn't create the file. "w+" creates
 				// the file and truncates it
-				f = fopen(fn.c_str(), "wb+");
+#ifdef TORRENT_WINDOWS
+				f = _wfopen(utf8_wchar(fn).c_str(), L"wb+");
+#else
+				f = _wfopen(fn.c_str(), "wb+");
```
Comment (arvidn) [inline src/posix_storage.cpp]: this is a typo, right?

Comment (arvidn) [inline src/posix_storage.cpp]:
```suggestion
				f = fopen(fn.c_str(), "wb+");
```

**Comment by arvidn** [comment]
the CI will fail, I'm working on fixing it. I will cancel it on travis for now

File: src/posix_storage.cpp (hunk 1 above)
Comment (pavel-pimenov) [reply src/posix_storage.cpp]: ups

**Comment by arvidn** [comment]
could you rebase this on top of master, so CI can run?

File: src/posix_storage.cpp [hunk 2]
```diff
@@ -512,10 +515,18 @@ namespace aux {
 	{
 		std::string const fn = files().file_path(idx, m_save_path);
 
-		char const* mode_str = (mode & open_mode::write)
+		const auto mode_str = (mode & open_mode::write)
```
Comment (arvidn) [inline src/posix_storage.cpp]:
```suggestion
		auto const* mode_str = (mode & open_mode::write)
```

right now the arrays decay to a pointer via the ternary operator. It should be explicit.

**Comment by arvidn** [comment]
please squash the commits when you rebase too

**Comment by pavel-pimenov** [comment]
create new PR

**Pull request closed without merging**